keyboards/skyloong/gk68/pro/pro.c

```diff
@@ -0,0 +1,233 @@
+    {0, CS10_SW11, CS11_SW11, CS12_SW11},
+    {0, CS10_SW12, CS11_SW12, CS12_SW12},
+    {0, CS18_SW9,  CS17_SW9,  CS16_SW9},
+    {0, CS18_SW10,  CS17_SW10,  CS16_SW10},
+
+    {0, CS13_SW1,  CS14_SW1,  CS15_SW1},
+    {0, CS13_SW2,  CS14_SW2,  CS15_SW2},
+    {0, CS13_SW3,  CS14_SW3,  CS15_SW3},
+    {0, CS13_SW5,  CS14_SW5,  CS15_SW5},
+    {0, CS13_SW6,  CS14_SW6,  CS15_SW6},
+    {0, CS13_SW8,  CS14_SW8,  CS15_SW8},
+    {0, CS13_SW9,  CS14_SW9,  CS15_SW9},
+    {0, CS13_SW10, CS14_SW10, CS15_SW10},
+    {0, CS13_SW11, CS14_SW11, CS15_SW11},
+    {0, CS13_SW12, CS14_SW12, CS15_SW12},
+    {0, CS18_SW11,  CS17_SW11,  CS16_SW11},
+    {0, CS18_SW12,  CS17_SW12,  CS16_SW12}
+};
+
+bool rgb_matrix_indicators_advanced_kb(uint8_t led_min, uint8_t led_max) {
+    if (!rgb_matrix_indicators_advanced_user(led_min, led_max)) {
+        return false;
+    }
+
+    if (host_keyboard_led_state().caps_lock) {
+        RGB_MATRIX_INDICATOR_SET_COLOR(CAPS_LOCK_INDEX, 255, 255, 255);
+    } else {
+        if (!rgb_matrix_get_flags()) {
+            RGB_MATRIX_INDICATOR_SET_COLOR(CAPS_LOCK_INDEX, 0, 0, 0);
+        }
+    }
+
+   switch (get_highest_layer(layer_state)) {
+      case 2:{
+        RGB_MATRIX_INDICATOR_SET_COLOR(WIN_MOD_INDEX, 255, 255, 255);
+        if (!rgb_matrix_get_flags()) {
+            RGB_MATRIX_INDICATOR_SET_COLOR(MAC_MOD_INDEX, 0, 0, 0);
+         }
+      } break;
+      case 3:{
+        RGB_MATRIX_INDICATOR_SET_COLOR(MAC_MOD_INDEX, 255, 255, 255);
+        if (!rgb_matrix_get_flags()) {
+            RGB_MATRIX_INDICATOR_SET_COLOR(WIN_MOD_INDEX, 0, 0, 0);
+         }
+      } break;
+
+      case 0:{
+       if (L_WIN) {
+            RGB_MATRIX_INDICATOR_SET_COLOR(WIN_MOD_INDEX, 255, 255, 255);
+            if (!rgb_matrix_get_flags()) {
+               RGB_MATRIX_INDICATOR_SET_COLOR(MAC_MOD_INDEX, 0, 0, 0);
+            }
+            }else{
+                if (!rgb_matrix_get_flags()) {
+                   RGB_MATRIX_INDICATOR_SET_COLOR(WIN_MOD_INDEX, 0, 0, 0);
+                 }
+              }
+         } break;
+
+     case 1:{
+       if (L_MAC) {
+            RGB_MATRIX_INDICATOR_SET_COLOR(MAC_MOD_INDEX, 255, 255, 255);
+            if (!rgb_matrix_get_flags()) {
+               RGB_MATRIX_INDICATOR_SET_COLOR(WIN_MOD_INDEX, 0, 0, 0);
+            }
+            }else{
+                if (!rgb_matrix_get_flags()) {
+                   RGB_MATRIX_INDICATOR_SET_COLOR(MAC_MOD_INDEX, 0, 0, 0);
+                 }
+              }
+         } break;
+
+      default:{
+         if (!rgb_matrix_get_flags()) {
+            RGB_MATRIX_INDICATOR_SET_COLOR(WIN_MOD_INDEX, 0, 0, 0);
+            RGB_MATRIX_INDICATOR_SET_COLOR(MAC_MOD_INDEX, 0, 0, 0);
+         }
+      }
+    }
+    return false;
+}
+
+
+void suspend_power_down_kb(void) {
+    writePinLow(SDB);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void) {
+    writePinHigh(SDB);
+    suspend_wakeup_init_user();
+}
+#endif
```

Comment:
kb shutdown function was recently added, and seems like it is prime for this: 
```suggestion
bool shutdown_kb(bool jump_to_bootloader) {
    writePinLow(SDB);
    return shutdown_user(jump_to_bootloader);
}
#endif
```